### PR TITLE
Move user docs to docs/user-guide/

### LINKED ISSUE #11093: Move user docs to docs/user-guide and create an up-to-date TOC
We've started to create a user guide in docs/user-guide. We should move related docs and concept-oriented examples to there, and create a TOC.

We probably need to merge, refactor, rename, etc. docs as well.

Files to move:
* docs/user-guide.md -> docs/user-guide/README.md, and updated
* Generated kubectl docs to docs/user-guide/kubectl/  @jlowdermilk 
* docs/accessing-the-cluster.md
* docs/annotations.md
* docs/application-troubleshooting.md
* docs/compute_resources.md
* docs/container-environment.md
* docs/containers.md
* docs/debugging-services.md
* docs/downward_api.md
* docs/identifiers.md
* docs/images.md
* docs/k8s-ui-*.png
* docs/kibana.png
* docs/kubeconfig-file.md
* docs/labels.md
* docs/logging.md
* docs/monitoring-architecture.png
* docs/monitoring.md
* docs/namespaces.md
* docs/overview.md
* docs/persistent-volumes.md
* docs/pod-states.md
* docs/pods.md
* docs/replication-controller.md
* docs/secrets.md
* docs/security_context.md
* docs/services.md
* docs/services*.png
* docs/sharing-clusters.md
* docs/ui.md
* docs/volumes.md
* docs/working_with_resources.md
* examples/downward-api
* examples/environment-guide
* examples/explorer
* examples/limitrange
* examples/liveness
* examples/logging-demo
* examples/node-selection
* examples/persistent-volumes
* examples/resourcequota
* examples/secrets
* examples/update-demo
* examples/walkthrough
* examples/multi-pod.yaml
* examples/pod.yaml
* examples/replication.yaml
* examples/simple-yaml.md
* examples/simple-nginx.md

Related: #9220, #9494 

cc @nikhiljindal @lavalamp

### PR DESCRIPTION
**Pull request by mikedanese**
Fixes #11093
@bgrant0607

### CODE REVIEW

File: docs/admin/namespaces.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ This is a placeholder document about namespace administration.
 TODO: document namespace creation, ownership assignment, visibility rules,
 policy creation, interaction with network.
 
-Namespaces are still under development.  For now, the best documentation is the [Namespaces Design Document](../design/namespaces.md).  The user documentation can be found at [Namespaces](../../docs/namespaces.md)
+Namespaces are still under development.  For now, the best documentation is the [Namespaces Design Document](../design/namespaces.md).  The user documentation can be found at [Namespaces](namespaces.md)
```
Comment (bgrant0607) [inline docs/admin/namespaces.md]: Should be ../../docs/user-guide/namespaces.md

Comment (mikedanese) [inline docs/admin/namespaces.md]: fixed.

**Comment by bgrant0607** [comment]
LGTM. Thanks!

**Comment by bgrant0607** [comment]
cc @erictune

**Comment by bgrant0607** [comment]
Need to update examples tests.

**Comment by bgrant0607** [comment]
cc @RichieEscarez

**Comment by mikedanese** [comment]
fixed tests

**Comment by bgrant0607** [comment]
Thanks. LGTM once Shippable passes.

**Pull request merged**